### PR TITLE
ENT-2678 | Excluded the users which have only audit enrollments if 'E…

### PR DESCRIPTION
**Pull request by hasnain-naveed**
…nable audit data reporting' is not enabled.

### CODE REVIEW

File: enterprise_data/filters.py [hunk 1]
```diff
@@ -85,3 +85,29 @@ def filter_queryset(self, request, queryset, view):
...
+        """
+        enterprise_id = view.kwargs['enterprise_id']
+
+        self.update_session_with_enterprise_data(request)
+
+        enable_audit_data_reporting = request.session['enable_audit_data_reporting'].get(enterprise_id, False)
+
+        enterprise_user_ids = []
+        if not enable_audit_data_reporting:
+            for record in queryset:
```
Comment (ziafazal) [inline enterprise_data/filters.py]: we should avoid this looping and use joins by filtering queryset
`queryset.filter(~Q(enrollments__user_current_enrollment_mode='audit')`

File: enterprise_data/models.py [hunk 2]
```diff
@@ -126,3 +126,21 @@ def __repr__(self):
         Return uniquely identifying string representation.
         """
         return self.__str__()
+
+    @property
+    def has_paid_mode_enrollment(self):
```
Comment (ziafazal) [inline enterprise_data/models.py]: This is not needed if make above queryset change.

**Review by ziafazal — CHANGES_REQUESTED** [review]
@hasnain-naveed completed first round of review.

**Review by ziafazal — APPROVED** [review]
👍

**Review by moconnell1453 — APPROVED** [review]

**Pull request merged**